meson.build

```diff
@@ -0,0 +1,157 @@
+project('milestonexprotectsrc', 'cpp',
+  version : '1.0.0',
+  meson_version : '>= 0.50',)
+
+gst_milestonexprotect_version = meson.project_version()
+version_arr = gst_milestonexprotect_version.split('.')
+gst_milestonexprotect_version_major = version_arr[0].to_int()
+gst_milestonexprotect_version_minor = version_arr[1].to_int()
+gst_milestonexprotect_version_micro = version_arr[2].to_int()
+if version_arr.length() == 4
+  gst_milestonexprotect_version_nano = version_arr[3].to_int()
+else
+  gst_milestonexprotect_version_nano = 0
+endif
+gst_milestonexprotect_version_is_dev = gst_milestonexprotect_version_minor % 2 == 1 and gst_milestonexprotect_version_micro < 90
+
+# Find external dependencies
+gst_req = '>=1.16.0'
+gst_app_dep = dependency('gstreamer-app-1.0', version : gst_req)
+gst_dep = dependency('gstreamer-1.0', version : gst_req,
+  fallback : ['gstreamer', 'gst_dep'])
+gstbase_dep = dependency('gstreamer-base-1.0', version : gst_req,
+  fallback : ['gstreamer', 'gst_base_dep'])
+glib_dep = dependency('glib-2.0')
+
+# Define compiler args and include directories
+gst_c_args = ['-DHAVE_CONFIG_H']
+configinc = include_directories('.')
+
+# Define gstreamer API version
+apiversion = '1.0'
+
+# Define installation directories
+prefix = get_option('prefix')
+plugin_install_dir = join_paths(get_option('libdir'), 'gstreamer-1.0')
+# Internal installation directory
+interpipe_plugin_dir = []
+
+# Get an object returns describing a compiler
+cc = meson.get_compiler('cpp')
+
+# Verify if the warning flags are available in the compiler
+# If the flags is availale for the compiler it wiil be used in all compiler
+# invocations with the exception of compile tests.
+warning_flags = [
+  '-Wmissing-declarations',
+  '-Wmissing-prototypes',
+  '-Wredundant-decls',
+  '-Wundef',
+  '-Wwrite-strings',
+  '-Wformat',
+  '-Wformat-nonliteral',
+  '-Wformat-security',
+  '-Wold-style-definition',
+  '-Winit-self',
+  '-Wmissing-include-dirs',
+  '-Waddress',
+  '-Waggregate-return',
+  '-Wno-multichar',
+  '-Wdeclaration-after-statement',
+  '-Wvla',
+  '-Wpointer-arith',
+]
+foreach extra_arg : warning_flags
+  if cc.has_argument (extra_arg)
```

Comment:
You can just use https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerget_supported_arguments here.

Reply:
Thanks again!